Last step of week 3.

ataddesse opened the pull request:
I was trying to modify the todo-list code example I followed on the walk-through. However, I was not able to fetch the data successfully. I could not see what is wrong with my code. Can you help me spot the error through this review?

ataddesse committed:
Last step of week 3.

anttsou-google commented:
What is the exact issue you are experiencing? It would be difficult to debug just from looking at this. Maybe you can add some logging and then see if the data is expected?

anttsou-google commented on portfolio/src/main/java/com/google/sps/servlets/ListCommentServlet.java:
```diff
@@ -0,0 +1,54 @@
+    response.setContentType("application/json;");
+    Query query = new Query("Messages").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
```
I've noticed you are not using gson to convert these to entity properties to json. That should happen before you do response.getWriter().println(). Not sure if that is the issue you are experiencing though.

ataddesse replied:
Sorry for the inconvenience on my previous commit. If I need to use gson, I am now trying to use ArrayList to temporarily store data for displaying. I will get back to you soon.

ataddesse committed:
Going back back to step 5 of week 3 to be fetch data from data store in a better way. This will help me to utilize the data efficiently.

ataddesse committed:
This commit is related with step 6 of week 3.
 - I have added a feature to let  the user select the number of comments displayed.

anttsou-google commented on portfolio/src/main/java/com/google/sps/servlets/ListCommentServlet.java:
```diff
@@ -11,43 +11,47 @@
-    String numResultsString = (request.getQueryString());
-    int reqResults = Integer.parseInt(numResultsString);
     
+    ArrayList<String> Comments = new ArrayList<String>();
```
use generic types when possible, so List<String> comments = new ArrayList<>();

ataddesse replied:
I will take a note of that.

anttsou-google commented on portfolio/src/main/java/com/google/sps/servlets/ListCommentServlet.java:
```diff
@@ -11,43 +11,47 @@
-    String numResultsString = (request.getQueryString());
-    int reqResults = Integer.parseInt(numResultsString);
     
+    ArrayList<String> Comments = new ArrayList<String>();
```
in java, we usually don't capitalize the first letter for variables, so "Comments" should be "comments"

ataddesse replied:
Alright, I will modify my code.

anttsou-google commented on portfolio/src/main/java/com/google/sps/servlets/ListCommentServlet.java:
```diff
@@ -11,43 +11,47 @@
-         if (reqResults <= 0) break;
+     
+      long id = entity.getKey().getId();
+      String comments = (String) entity.getProperty("comment");
```
nit: since it is a single comment, should this be "String comment" for more clarity?

ataddesse replied:
Yes, I agree.

anttsou-google commented on portfolio/src/main/java/com/google/sps/servlets/ListCommentServlet.java:
```diff
@@ -11,43 +11,47 @@
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      
     response.setContentType("text/html;");
     response.setContentType("application/json;");
```
are these necessary if you setContentType("application/json;") already at the bottom?

ataddesse replied:
Yes, that should be removed. I forgot to remove it after editing my previous code.

anttsou-google commented on portfolio/src/main/webapp/script.js:
```diff
@@ -87,43 +78,32 @@ function createListElement(text) {
+  fetch('/list-post').then(response => response.json()).then((comment) => {
+    const statsListElement = document.getElementById('post-list');
+     statsListElement.innerHTML = '';
+    for (i = 0; i < limitParsed; i++) {
```
i'm surprised this for loop works, as you usually need to initialize "i" with "let i" or "var i". Is this behaving as intended?

ataddesse replied:
I accidentally did not initialize it, it was working well though. Anyways, I will modify my code to follow rules and avoid bugs.

anttsou-google commented on portfolio/src/main/java/com/google/sps/servlets/ListCommentServlet.java:
```diff
@@ -11,43 +11,47 @@
+      Comments.add(comments);
+    }
 
+    for(String mess : Comments){
```
remove logging in production code

ataddesse replied:
Alright!

anttsou-google commented on portfolio/src/main/java/com/google/sps/servlets/ListCommentServlet.java:
```diff
@@ -11,43 +11,47 @@
+    Gson gson = new Gson();
 
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(Comments));
```
you can just inline this to "new Gson().toJson(Comments));"

ataddesse replied:
That will be cool.

anttsou-google reviewed (commented):
have a few comments/suggestions. Also, please try to fix the spacing before merging code to master. Thanks!

ataddesse committed:
In this commit, I have added a Delete all feature for the user, including a button UI. Besides this, I have resolved previous code review comments.

anttsou-google commented on portfolio/src/main/webapp/script.js:
```diff
@@ -82,3 +73,43 @@ function createListElement(text) {
+
+    //Output every comment until reaching the limit
+    var i;
+    for (i = 0; i < difference; i++) {
```
you should be able to put the initialization of "i" directly in the for loop

ataddesse replied:
Alright.

anttsou-google reviewed (approved):
LGTM
